File: 3-Patron/python/CreadorMapaLava.py
import CreadorDeMapa
import Textura
import Mapa


class CreadorMapaLava(CreadorDeMapa):
    @property
    def mapa(self):
        return self.__mapa

    def crearMapa(self):
        mazmorra = None
        tesoro = None
        textura = Textura("nieve.png", 100, 200)
        self.__mapa = Mapa(mazmorra, tesoro, textura)
        return self.__mapa

File: 3-Patron/php/CreadorMapaLava.php
<?php
include './CreadorDeMapa.php';
include './Mapa.php';
include './Mazmorra.php';
include './Textura.php';
include './Tesoro.php';

class CreadorMapaLava extends CreadorDeMapa{
	public function __construct($mapa){
    	parent::__construct($mapa);
	}
	public function crearMapa(){
	    $mazmorra = new Mazmorra();
	    $textura = new Textura('lava.png',140,300);
	    $tesoro = new Tesoro(100,2000);
	    $mapa = new Mapa($mazmorra,$tesoro,$textura);
	    return $mapa;
	}

	public function getMapa(){
		return parent::getMapa();	
	}

	public function setMapa($mapa){
		return parent::setMapa($mapa);
	}
}
?>
